kasa/tests/test_aestransport.py

```diff
@@ -285,6 +286,68 @@ async def test_port_override():
     assert str(transport._app_url) == "http://127.0.0.1:12345/app"
 
 
+@pytest.mark.parametrize(
+    "request_delay, should_error, should_succeed",
+    [(0, False, True), (0.125, True, True), (0.3, True, True), (0.7, True, False)],
+    ids=["No error", "Error then succeed", "Two errors then succeed", "No succeed"],
+)
+async def test_device_closes_connection(
+    mocker, request_delay, should_error, should_succeed
+):
+    """Test the delay logic in http client to deal with devices that close connections after each request.
+
+    Currently only the P100 on older firmware.
+    """
+    host = "127.0.0.1"
+
+    # Speed up the test by dividing all times by a factor. Doesn't seem to work on windows
+    # but leaving here as a TODO to manipulate system time for testing.
+    speed_up_factor = 1
+    default_delay = HttpClient.WAIT_BETWEEN_REQUESTS_ON_OSERROR / speed_up_factor
+    request_delay = request_delay / speed_up_factor
+    mock_aes_device = MockAesDevice(
+        host, 200, 0, 0, sequential_request_delay=request_delay
+    )
+    mocker.patch.object(aiohttp.ClientSession, "post", side_effect=mock_aes_device.post)
+
+    config = DeviceConfig(host, credentials=Credentials("foo", "bar"))
+    transport = AesTransport(config=config)
+    transport._http_client.WAIT_BETWEEN_REQUESTS_ON_OSERROR = default_delay
+    transport._state = TransportState.LOGIN_REQUIRED
+    transport._session_expire_at = time.time() + 86400
```

Comment:
I'm not so sure we should use time.time() inside tests at all. IIRC, freezegun and other similar are generally recommended for mocking time related functions.

Reply:
Yes I agree and I think there are other places where this can speed up tests

Reply:
Should that be done in this PR, or should we just merge this right now?

Reply:
I think let's merge and then we've fixed the P100, and I'll look at freeze gun next across the board for all our tests.